core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -609,7 +609,7 @@ private void createImpliedDependencies(QueryModel parent) {
         JoinContext jc;
         for (int i = 0, n = models.size(); i < n; i++) {
             QueryModel m = models.getQuick(i);
-            if (m.getJoinType() == QueryModel.JOIN_ASOF || m.getJoinType() == QueryModel.JOIN_SPLICE) {
+            if (m.getJoinType() == QueryModel.JOIN_ASOF || m.getJoinType() == QueryModel.JOIN_SPLICE || m.getJoinType() == QueryModel.JOIN_LT) {
```

Comment:
another list lookup, also feels that for implied timestamp we should use same joins that participate in `requiredTimestamp` just above. Do you mind checking why CROSS join isn't present here?